src/components/functions/forms/GenericForm.js

```diff
@@ -11,8 +11,8 @@ const FormItem = Form.Item;
 const InputGroup = Input.Group;
 const Option = Select.Option;
 
-const floatPattern = /^\d+\.?\d*$/;
-const intPattern = /^\d+$/;
+const floatPattern = /^[+-]?\d+\.?\d*([eE][+-]?\d+)?$/;
+const intPattern = /^\d+([eE][+]?\d+)?$/;
```

Comment:
Does it match negative integers ?

Reply:
I've made some tests here https://regex101.com/ and it seems not.

Reply:
It does not, because for now, all physical parameters are floats and negative integers would allow creating negative number of points at plot, which makes no sense. Of course if there would be any physical int parameter then it would be necessary to resolve this problem other way, but for now this should work.

Reply:
If this is by purpose, then what about renaming the variable to `uintPattern` and adding a comment that it matches unsigned integers ?
Please add comment on `floatPattern` explaining what type of expressions would match the regexp. It is much easier for developer to read a comment than to decipher regexp